src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -80,6 +83,28 @@ private static byte[] EncodeLength(int length)
         }
 
         /// <summary>
+        /// Encode the segments { tag, length, value } of a boolean.
+        /// </summary>
+        /// <param name="value">The boolean to encode</param>
+        /// <returns>The encoded segments { tag, length, value }</returns>
+        internal static byte[][] SegmentedEncodeBoolean(bool value)
+        {
+            // BER says FALSE is zero, TRUE is other.
+            // DER says TRUE is 0xFF.
+            byte[] data =
+            {
+                (byte)(value ? 0xFF : 0x00),
+            };
+
+            return new byte[][]
+            {
+                new byte[] { (byte)DerSequenceReader.DerTag.Boolean }, 
+                new byte[] { 0x01 }, 
```

Comment:
I'd like to take zeroth credit:

> The byte[][] Segmented helpers are reducing the amount of memory copying, but also kind of a pain. I considered trying to make rich types for them in the middle of this change, but if we can avoid too much more I think still that's better suited to public API (post-1.0.0-rtm).

Since right now it's a source-include I'm trying to keep it light; but maybe some nested types are called for.